Sources/Kitura/RouterResponse.swift

```diff
@@ -549,14 +425,173 @@ public class RouterResponse {
+    /// Send data.
+    ///
+    /// - Parameter data: The data to send.
+    /// - Returns: This RouterResponse.
+    @discardableResult
+    public func send(data: Data) -> RouterResponse {
+        guard !state.invokedEnd else {
+            Log.warning("RouterResponse send(data:) invoked after end() for \(self.request.urlURL)")
+            return self
+        }
+        buffer.append(data: data)
+        state.invokedSend = true
+        return self
+    }
+    
+    /// Send a file.
+    ///
+    /// - Parameter fileName: The name of the file to send.
+    /// - Throws: An error in the Cocoa domain, if the file cannot be read.
+    /// - Returns: This RouterResponse.
+    ///
+    /// - Note: Sets the Content-Type header based on the "extension" of the file.
+    ///       If the fileName is relative, it is relative to the current directory.
+    @discardableResult
+    public func send(fileName: String) throws -> RouterResponse {
+        guard !state.invokedEnd else {
+            Log.warning("RouterResponse send(fileName:) invoked after end() for \(self.request.urlURL)")
+            return self
+        }
+        let data = try Data(contentsOf: URL(fileURLWithPath: fileName))
+        
+        let contentType = ContentType.sharedInstance.getContentType(forFileName: fileName)
+        if  let contentType = contentType {
+            headers["Content-Type"] = contentType
+        }
+        
+        send(data: data)
+        
+        return self
+    }
+    
+    /// Set headers and attach file for downloading.
+    ///
+    /// - Parameter download: The file to download.
+    /// - Throws: An error in the Cocoa domain, if the file cannot be read.
+    public func send(download: String) throws {
+        guard !state.invokedEnd else {
+            Log.warning("RouterResponse send(download:) invoked after end() for \(self.request.urlURL)")
+            return
+        }
+        try send(fileName: StaticFileServer.ResourcePathHandler.getAbsolutePath(for: download))
+        headers.addAttachment(for: download)
+    }
+    
+    typealias JSONSerializationType = JSONSerialization
+    
+    /// Send JSON.
+    ///
+    /// - Parameter json: The array to send in JSON format.
+    /// - Returns: This RouterResponse.
+    @discardableResult
+    public func send(json: [Any]) -> RouterResponse {
+        guard !state.invokedEnd else {
+            Log.warning("RouterResponse send(json:) invoked after end() for \(self.request.urlURL)")
+            return self
+        }
+        
+        do {
+            let jsonData = try JSONSerializationType.data(withJSONObject: json, options:.prettyPrinted)
+            headers.setType("json")
+            send(data: jsonData)
+        } catch {
+            Log.warning("Failed to convert JSON for sending: \(error.localizedDescription)")
+        }
+        
+        return self
+    }
+    
+    /// Send JSON.
+    ///
+    /// - Parameter json: The Dictionary to send in JSON format as a hash.
```

Comment:
Similar comment to the previous function, also I'm not sure what "as a hash" means.  Perhaps that could just be omitted.